core/shared/platform/linux-sgx/sgx_socket.c

```diff
@@ -237,6 +249,46 @@ textual_addr_to_sockaddr(const char *textual, int port, struct sockaddr_in *out)
     return BHT_OK;
 }
 
+static int
+sockaddr_to_bh_sockaddr(const struct sockaddr *sockaddr, socklen_t socklen,
```

Comment:
this is a clone of a function that we have in the `posix_socket.c` - do you think we can share the code instead of replicating it?

Reply:
Yes, maybe we add folder `core/shared/platform/common/socket` and a file under it to implement `sockaddr_to_bh_sockaddr` and `bh_sockaddr_to_sockaddr`. But here SGX only supports `AF_INET` and `bh_sockaddr->is_ipv4`, the code is a little different from posix's implementation, I think we can merge this PR firstly and refine the code later.